### PR TITLE
Update GKE version extraction/normalization

### PR DESCRIPTION
**Pull request by liggitt**
Follow up to https://github.com/kubernetes-sigs/kubetest2/pull/70 with additional tests/matchers

/cc @amwat

### CODE REVIEW

File: kubetest2-gke/deployer/build_test.go [hunk 1]
```diff
@@ -70,7 +112,7 @@ func TestNormalizeVersion(t *testing.T) {
 		{
 			name:            "alpha version with patch",
 			version:         "1.16.13-alpha.123",
-			expectedVersion: "1.16.13-gke.123+alpha",
+			expectedVersion: "1.16.13-gke.99.0+alpha.123",
```
Comment (amwat) [inline kubetest2-gke/deployer/build_test.go:115]: This behavior is changing, are we sure we don't want `.123.0` instead?

**Review by amwat — COMMENTED** [review]
one question, lgtm otherwise

/lgtm
/approve
/hold

File: kubetest2-gke/deployer/build_test.go (hunk 1 above)
Comment (liggitt) [reply kubetest2-gke/deployer/build_test.go:115]: yeah, alpha.123 has no real bearing on gke build version... preserving the existing suffix intact makes more sense to me

Comment (amwat) [reply kubetest2-gke/deployer/build_test.go:115]: sgtm

/hold cancel

**Pull request merged**